### DISCUSSION TITLE
better late than never - Day 7!

### DISCUSSION THREAD

**Discussion by CarolineMorton** [CarolineMorton/AdventOfCode_WIR#10] [Ideas]
I had a big family party this weekend so was out much of the weekend!

Getting back to puzzling now - how are you all getting on?

**Comment by kuhmil** [comment]
This is my attempt of [Part 1 Day 07](https://github.com/kuhmil/AdventOfCode_WIR/tree/main/day_07). I cannot figure out what is wrong with my logic as this works with the test input. Maybe I should focus on writing tests for this?

**Reply by CarolineMorton** [reply]
Yes i think tests can help you out here. If you pull [this](https://github.com/kuhmil/AdventOfCode_WIR/blob/027588b9880ea623dda0c464f2416aeaa6ded2b4/day_07/src/main.rs#L32) into its own file, then main.rs won't get cluttered with tests.

**Comment by em-baggie** [comment]
<a href="https://github.com/em-baggie/AdventOfCode_WIR/tree/main/day_07">My attempt</a> at day 7 pt 1. I get the correct answer if the input is around 200 lines or less, but becomes inaccurate if input increased >205 lines. Not sure why, thought it might be due to overflow, but changing everything to i64 didn't solve it. Anyone got any ideas???

EDIT: I worked out why this was happening! The only duplicate entry in the input is with key of 894. I was parsing the input into a hashmap and this was not including both entries, so my answer was 894 less than it should have been. Switched to parsing the input to vector and now it works!

**Reply by CarolineMorton** [reply]
Well done!

**Comment by Sowmiya07** [comment]
[Here](https://github.com/Sowmiya07/AdventOfCode_WIR/blob/main/day_07/src/main.rs) is my solution for day 7 part 1.
